### ISSUE TITLE
compile error

### ISSUE THREAD

**Issue by FenHua** [lilanxiao/Rotated_IoU#10]
can you help me？I meet some mistakes

cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
In file included from sort_vert.cpp:1:0:
utils.h:27:64: warning: backslash-newline at end of file
 #define CHECK_IS_BOOL(x)                                       \
                                                                ^
sort_vert.cpp: In function ‘at::Tensor sort_vertices(at::Tensor, at::Tensor, at::Tensor)’:
utils.h:12:73: error: ‘TORCH_CHECK’ was not declared in this scope
         TORCH_CHECK(x.is_contiguous(), #x " must ne a contiguous tensor");       \
                                                                         ^
sort_vert.cpp:7:5: note: in expansion of macro ‘CHECK_CONTIGUOUS’
     CHECK_CONTIGUOUS(vertices);
     ^
utils.h:12:73: error: ‘TORCH_CHECK’ was not declared in this scope
         TORCH_CHECK(x.is_contiguous(), #x " must ne a contiguous tensor");       \
                                                                         ^
sort_vert.cpp:8:5: note: in expansion of macro ‘CHECK_CONTIGUOUS’
     CHECK_CONTIGUOUS(mask);
     ^
utils.h:12:73: error: ‘TORCH_CHECK’ was not declared in this scope
         TORCH_CHECK(x.is_contiguous(), #x " must ne a contiguous tensor");       \
                                                                         ^
sort_vert.cpp:9:5: note: in expansion of macro ‘CHECK_CONTIGUOUS’
     CHECK_CONTIGUOUS(num_valid);
     ^
utils.h:7:61: error: ‘TORCH_CHECK’ was not declared in this scope
         TORCH_CHECK(x.is_cuda(), #x " must be a CUDA tensor");   \
                                                             ^
sort_vert.cpp:10:5: note: in expansion of macro ‘CHECK_CUDA’
     CHECK_CUDA(vertices);
     ^
utils.h:7:61: error: ‘TORCH_CHECK’ was not declared in this scope
         TORCH_CHECK(x.is_cuda(), #x " must be a CUDA tensor");   \
                                                             ^
sort_vert.cpp:11:5: note: in expansion of macro ‘CHECK_CUDA’
     CHECK_CUDA(mask);
     ^
utils.h:7:61: error: ‘TORCH_CHECK’ was not declared in this scope
         TORCH_CHECK(x.is_cuda(), #x " must be a CUDA tensor");   \
                                                             ^
sort_vert.cpp:12:5: note: in expansion of macro ‘CHECK_CUDA’
     CHECK_CUDA(num_valid);
     ^
utils.h:24:49: error: ‘TORCH_CHECK’ was not declared in this scope
                     #x " must be a float tensor");              \
                                                 ^
sort_vert.cpp:13:5: note: in expansion of macro ‘CHECK_IS_FLOAT’
     CHECK_IS_FLOAT(vertices);
     ^
utils.h:30:48: error: ‘TORCH_CHECK’ was not declared in this scope
                     #x " must be a bool tensor");             \
                                                ^
sort_vert.cpp:14:5: note: in expansion of macro ‘CHECK_IS_BOOL’
     CHECK_IS_BOOL(mask);
     ^
utils.h:18:47: error: ‘TORCH_CHECK’ was not declared in this scope
                     #x " must be a int tensor");            \
                                               ^
sort_vert.cpp:15:5: note: in expansion of macro ‘CHECK_IS_INT’
     CHECK_IS_INT(num_valid);
     ^
sort_vert.cpp:23:54: error: expected primary-expression before ‘float’
     sort_vertices_wrapper(b, n, m, vertices.data_ptr<float>(), mask.data_ptr<bool>
                                                      ^
sort_vert.cpp:23:78: error: expected primary-expression before ‘bool’
  sort_vertices_wrapper(b, n, m, vertices.data_ptr<float>(), mask.data_ptr<bool>(),
                                                                           ^
sort_vert.cpp:24:45: error: expected primary-expression before ‘int’
                          num_valid.data_ptr<int>(), idx.data_ptr<int>());
                                             ^
sort_vert.cpp:24:66: error: expected primary-expression before ‘int’
                          num_valid.data_ptr<int>(), idx.data_ptr<int>());
                                                                  ^
error: command 'gcc' failed with exit status 1

**Comment by lilanxiao** [comment]
em... I think the problem comes from this error:

> utils.h:12:73: error: ‘TORCH_CHECK’ was not declared in this scope

TORCH_CHECK should be a macro provided by Pytorch and is only available in relatively newer versions. So I think your environment probably doesn't have the proper version of Pytorch. Can you check the version of Pytorch by running this in a new script? 

```
import torch
print(torch.__version__)
```
I've only tested my code with Pytorch 1.5. So if your version is older than 1.5, just install a 1.5 version and try again.
Hope it would help.

**Comment by FenHua** [comment]
> em... I think the problem comes from this error:
> 
> > utils.h:12:73: error: ‘TORCH_CHECK’ was not declared in this scope
> 
> TORCH_CHECK should be a macro provided by Pytorch and is only available in relatively newer versions. So I think your environment probably doesn't have the proper version of Pytorch. Can you check the version of Pytorch by running this in a new script?
> 
> ```
> import torch
> print(torch.__version__)
> ```
> 
> I've only tested my code with Pytorch 1.5. So if your version is older than 1.5, just install a 1.5 version and try again.
> Hope it would help.

Thank you very much, I have solved it by changing Pytorch 1.6 to Pytorch 1.2 ~~
